### PR TITLE
Adding required tables for the new cookbook_artifacts endpoint

### PR DESCRIPTION
**Pull request by wk8**
Mostly similar to the good old 'cookbook*' tables, but still different
enough to make them separate.
Also makes sense since we want this new endpoint's data completely
separate from the old one's.

### CODE REVIEW

**Comment by danielsdeleo** [comment]
My SQL is rather rusty but this looks good to me.

File: deploy/cookbook_artifact_versions.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  metadata bytea NOT NULL,
+  serialized_object bytea NOT NULL,
+  updated_at TIMESTAMP WITHOUT TIME ZONE NOT NULL,
+  created_at TIMESTAMP WITHOUT TIME ZONE NOT NULL,
+  last_updated_by CHAR(32) NOT NULL,
+  cookbook_id INTEGER NOT NULL REFERENCES cookbooks(id) ON DELETE RESTRICT,
+  UNIQUE(cookbook_id, identifier)
+);
+
+ALTER TABLE cookbook_artifact_versions ALTER meta_attributes SET STORAGE EXTERNAL;
```
Comment (marcparadise) [inline deploy/cookbook_artifact_versions.sql]: Not familiar with EXTERNAL, what's it do?

Comment (seth) [inline deploy/cookbook_artifact_versions.sql]: I believe EXTERNAL controls whether or not pg does compression on the data (although that's probably not all it means).

Comment (wk8) [inline deploy/cookbook_artifact_versions.sql]: From http://www.postgresql.org/docs/9.2/static/sql-altertable.html : 
```
SET STORAGE
This form sets the storage mode for a column. This controls whether this column is held inline or in a secondary TOAST table, and whether the data should be compressed or not. PLAIN must be used for fixed-length values such as integer and is inline, uncompressed. MAIN is for inline, compressible data. EXTERNAL is for external, uncompressed data, and EXTENDED is for external, compressed data. EXTENDED is the default for most data types that support non-PLAIN storage. Use of EXTERNAL will make substring operations on very large text and bytea values run faster, at the penalty of increased storage space. Note that SET STORAGE doesn't itself change anything in the table, it just sets the strategy to be pursued during future table updates. See Section 56.2 for more information.
```

**Comment by wk8** [comment]
Approved by @marcparadise on hipchat, merging :-)

**Pull request merged**